lib/internal/url.js

```diff
@@ -228,14 +229,14 @@ class URLSearchParams {
             const typedKey = toUSVString(key);
             const typedValue = toUSVString(init[key]);
 
-            // Two different key may result same after `toUSVString()`, we only
-            // leave the later one. Refers to WPT.
-            if (visited[typedKey] !== undefined) {
-              this[searchParams][visited[typedKey]] = typedValue;
+            // Two different keys may become the same USVString after normalization.
+            // In that case, we retain the later one. Refer to WPT.
+            if (visited.has(typedKey)) {
+              this[searchParams][visited.get(typedKey)] = typedValue;
```

Comment:
```suggestion
            const visitedKey = visited.get(typedKey);
            if (visitedKey !== undefined) {
              this[searchParams][visitedKey] = typedValue;
```

Reply:
The revised code makes it clear that in `this[searchParams][visitedKey]`, the value `visitedKey` is not `undefined`. It seems conceptually better to me. It is probably slightly more maintainable... no?

Reply:
@Trott There's no point in doing a lookup twice.

Reply:
Used a slightly different variable name.

Reply:
> It is probably slightly more maintainable... no?

Yes, it seems more maintainable and easier to understand. I wanted to know if there were other motivations, but in retrospect, it doesn't really matter.